README.md

```diff
@@ -2217,4 +2217,88 @@ A Japanese word meaning "form", relating to karate. Used commonly in programming
 
 An excellent example of a Test Driven Development Kata:
 
-[The Bowling Game](http://butunclebob.com/ArticleS.UncleBob.TheBowlingGameKata)
\ No newline at end of file
+[The Bowling Game](http://butunclebob.com/ArticleS.UncleBob.TheBowlingGameKata)
+
+## Coding in the Command Line
+
+The Command Line is an example of a Read-Evaluate-Print-Loop (REPL). Users can enter expressions which will be read, evaluated and the answer printed to the screen. The environment will then return to the read state. This provides the ability to enter single expressions in different programming languages (once you have installed the program and called it from the command line). This ability can be useful for testing simple concepts or debugging.
+
+For example, if you write `python` into the command line and press enter you can then write the following:
+
+``` python
+def fib(n):
+    if n<3:
+        return n
+    return fib(n-1) + fib(n-2)
+```
+
+Getting the indentations right is very important as Python does not have brackets for if statements.
+
+You can then enter `fib(4)` and the command line will return `5`, or `fib(8)` and `34` will be returned.
+
+## Some different coding languages and their usages
+
+### MatLab (Matrix Laboratory)
+Good for matrices and vectors, which are the key components of linear algebra.
+
+### R
+Good for Statistics.
+
+### Python
+Good for anything, but especially useful for Machine Learning. Python had a major release to 3.0 at the end of 2008, but some software still uses 2.6 or 2.7 as it cannot handle the breaking changes.
+
+### Java
+Good for anything. Originally comes from Oracle Java, but was taken by developers and made into the open-source Java Development Kit (JDK). Oracle Java is almost the same as JDK.
```

Comment:
That's not quite right:

It's "Open JDK" (that's the name) vs. the "Oracle JDK". It's also worth noting that as of Java 1.11, there's only one JDK.